packages/devextreme/js/__internal/core/state_manager/state_manager.ts

```diff
@@ -0,0 +1,199 @@
+        }
+
+        const isObject = propertyValue !== undefined && propertyValue !== null
+          && typeof propertyValue === 'object';
+        this.componentState[preparedSourceDataId][propertyName] = isObject
+          ? new WeakRef(propertyValue) : propertyValue;
+
+        this.trackStateSourceChanges(
+          preparedSourceDataId,
+          propertyName,
+          propertyValue,
+        );
+      } else {
+        this.logger.debug(`No value container manager found for the '${propertyName}' property of the '${preparedSourceDataId}' state source`);
+      }
+    });
+  }
+
+  private trackStateSourceChanges(
+    stateId: string,
+    propertyName: string,
+    propertyValue: StateManagementTypes.MaybeValueContainer,
+  ): void {
+    const valueContainerManager = this.findValueContainerManagerFor(propertyValue);
+
+    if (!valueContainerManager) {
+      this.logger.debug(`No value container manager found for the '${propertyName}' property of the '${stateId}' state`);
+      return;
+    }
+
+    const fullPathToProperty = joinStatePath(stateId, propertyName);
+
+    try {
+      valueContainerManager.trackChanges(
+        (valueContainerChange: StateManagementTypes.ValueContainerChange) => {
+          const valueContainerChangeCopy = {
+            ...valueContainerChange,
+            payload: { ...valueContainerChange.payload, path: fullPathToProperty },
+          };
+
+          const { previousValue, newValue } = valueContainerChange.payload;
+
+          if (typeof previousValue === 'object' && previousValue !== null) {
+            valueContainerChangeCopy.payload.previousValue = deepCopy(previousValue);
+          }
+
+          if (typeof newValue === 'object' && newValue !== null) {
+            valueContainerChangeCopy.payload.newValue = deepCopy(newValue);
+          }
+
+          const updatedComponentState = this.getComponentState();
+
+          this.devToolsConnector
+            .sendAction(
+              'UPDATE',
+              valueContainerChangeCopy.payload,
+              updatedComponentState,
+            );
+        },
+      );
+    } catch (error) {
+      this.logger.error(`Failed to track state for ${fullPathToProperty}`, error);
+    }
+  }
+
+  private hasValueContainerManagerFor(
+    valueContainer: StateManagementTypes.MaybeValueContainer,
+  ): boolean {
+    return this.valueContainerManagers
+      // eslint-disable-next-line @stylistic/max-len
+      .some((currentStateContainerManager) => currentStateContainerManager.canHandle(valueContainer));
+  }
+
+  private findValueContainerManagerFor(
+    valueContainer: StateManagementTypes.MaybeValueContainer,
+  ): StateManagementTypes.ValueContainerManager | undefined {
+    const valueContainerManagerFactory = this.valueContainerManagers
+      // eslint-disable-next-line @stylistic/max-len
+      .find((currentStateContainerManager) => currentStateContainerManager.canHandle(valueContainer));
```

Comment:
Let's fix linter error instead of disabling it

```suggestion
      .find((
        currentStateContainerManager,
      ) => currentStateContainerManager.canHandle(valueContainer));
```

Reply:
Changed